### PR TITLE
Fix some issue with Windows

### LINKED ISSUE #339: When `exit_code` or `run` is invoked, the task does not complete (`fwrite()` error)
Castor 0.14.0 (Phar)
PHP 8.3.4
PowerShell 7.4.1 on Windows 11

Everytime I try to run a task that includes a call to `run()` or `exit_code()`, the task does not complete. I have to either `Ctrl+C` or press `Enter` to exit the task, and then an error is displayed:

```
In AbstractPipes.php line 139:

  Notice: fwrite(): Write of 2 bytes failed with errno=22 Invalid argument
```

The command is OK, because I can see its output right after this error message.
A sample task triggering this error:

```php
#[AsTask]
function phpVersion(): void {
  run("php --version"); // Using a string or an array as argument does not change anything.
}
```

This error does not occur if I use Symfony Process directly (i.e. `(new Process(["php", "--version"]))->run()`).

### PR DESCRIPTION
**Pull request by lyrixx**
fix #339
fix #342

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,10 @@
 * Find root directory by looking for a `.castor/castor.php` file
 * Allow stub file to be in `.castor/.castor.stub.php`
 
+* Fix issue with PTY on windows, it's not always disabled
```
Comment (pyrech) [inline CHANGELOG.md:19]:
```suggestion
* Fix issue with PTY on windows, it's now always disabled
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,10 @@
...
+* Fix issue when finding root dir on windows
+* Fix issue SymfonyTask creation
```
Comment (pyrech) [inline CHANGELOG.md:21]:
```suggestion
* Fix issue on SymfonyTask creation
```

File: examples/run.php [hunk 3]
```diff
@@ -15,7 +15,11 @@
 #[AsTask(description: 'Run a sub-process and display information about it')]
 function ls(): void
 {
-    $process = run('ls -alh && echo $foo', quiet: true, environment: ['foo' => 'ba\'"`r']);
+    if ('\\' === \DIRECTORY_SEPARATOR) {
```
Comment (pyrech) [inline examples/run.php:18]: Use `OsHelper::isWindows` instead

File: src/functions.php [hunk 4]
```diff
@@ -128,12 +128,15 @@ function run(
...
     }
 
-    if ($context->tty) {
-        $process->setTty(true);
-        $process->setInput(\STDIN);
-    } elseif ($context->pty) {
-        $process->setPty(true);
-        $process->setInput(\STDIN);
+    // TTY does not work on windows, and PTY is a mess, so let's skip everything!
+    if ('\\' !== \DIRECTORY_SEPARATOR) {
```
Comment (pyrech) [inline src/functions.php:132]: ! OsHelper::isWindows()

**Review by pyrech — CHANGES_REQUESTED** [review]

**Comment by lyrixx** [comment]
It would be nice indeed

**Review by pyrech — APPROVED** [review]
LGTM :sparkles:

**Pull request merged**